go/vt/vtgate/planbuilder/dml.go

```diff
@@ -36,34 +36,53 @@ func dmlFormatter(buf *sqlparser.TrackedBuffer, node sqlparser.SQLNode) {
 
 // buildUpdatePlan builds the instructions for an UPDATE statement.
 func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, error) {
-	route := &engine.Route{
+	er := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok || len(upd.TableExprs) != 1 {
+		bldr, err := processTableExprs(upd.TableExprs, vschema)
+		if err != nil {
+			return nil, err
+		}
+		rb, ok := bldr.(*route)
+		if !ok || rb.ERoute.Keyspace.Sharded {
+			return nil, errors.New("unsupported: multi-table update statement in sharded keyspace")
+		}
+		if hasSubquery(upd) {
+			return nil, errors.New("unsupported: subqueries in DML")
+		}
+		er.Keyspace = rb.ERoute.Keyspace
+		er.Opcode = engine.UpdateUnsharded
+		return er, nil
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
```

Comment:
`Table` is used in `getDMLRouting`. So I pulled the table from the route's symtab.